### PR TITLE
fix(all): remove all un-used code

### PR DESCRIPTION
**Pull request by anthony-liddle**
## Goal

This removes _all_ unused code bits. Plus the twitter integration.

## Todos:
- [x] Comb through the code and remove anything unused
- [x] Drop twitter integration

## Implementation Decisions
giphy

### CODE REVIEW

File: src/common/api/saving/remove.js [hunk 1]
```diff
@@ -2,15 +2,15 @@ import { request } from '../_request/request'
...
+export function removeItem(itemId) {
   return request({
     path: 'send/',
     data: {
       actions: [
         {
           action: 'delete',
           item_id: itemId,
-          ...actionInfo
+          type: 'page'
```
Comment (anthony-liddle) [inline src/common/api/saving/remove.js:13]: We don't need to pass in any extra info. This was hardcoded as `type: page` in the previous version of the extension 😂

File: src/actions.js [hunk 2]
```diff
@@ -46,18 +25,4 @@ export const TAG_SYNC_FAILURE = 'TAG_SYNC_FAILURE'
...
-export const USER_LOG_OUT_FAILURE = 'USER_LOG_IN_FAILURE'
-
-export const CHECK_TWITTER_INTEGRATION = 'CHECK_TWITTER_INTEGRATION'
-export const START_TWITTER_INTEGRATION = 'START_TWITTER_INTEGRATION'
-export const STOP_TWITTER_INTEGRATION = 'STOP_TWITTER_INTEGRATION'
-export const SAVE_TWEET_TO_POCKET = 'SAVE_TWEET_TO_POCKET'
-
-export const TWITTER_SAVE_REQUEST = 'TWITTER_SAVE_REQUEST'
-export const TWITTER_SAVE_SUCCESS = 'TWITTER_SAVE_SUCCESS'
-export const TWITTER_SAVE_FAILURE = 'TWITTER_SAVE_FAILURE'
```
Comment (anthony-liddle) [inline src/actions.js:63]: I did a Find All and none of these came up 🧹

File: src/common/api/index.js [hunk 3]
```diff
@@ -1,23 +1,17 @@
...
   getOnSaveTags,
   request,
-  // archiveItem,
   removeItem,
   getGuid,
-  sendAnalytics,
   syncItemTags,
-  fetchStoredTags,
-  getFeatures
+  fetchStoredTags
```
Comment (anthony-liddle) [inline src/common/api/index.js:16]: We weren't using any of these old API calls 🧹

File: src/common/constants.js [hunk 4]
```diff
@@ -1,8 +1,3 @@
-export const CACHE_INTERVAL = 900000
-export const KEY_ENTER = 13
-export const KEY_ESCAPE = 27
-export const LINK_HELP_PAGE =
-  'https://help.getpocket.com/article/1141-using-the-pocket-new-tab-extension-in-chrome'
```
Comment (anthony-liddle) [inline src/common/constants.js:5]: Not using these anywhere 🧹

File: src/common/helpers.js [hunk 5]
```diff
@@ -79,21 +41,6 @@ export function getCurrentLanguageCode() {
...
-
-export function getImageCacheUrl(url, imageSize) {
-  if (!url) return
-  const resizeParam = imageSize ? `${imageSize.width}x${imageSize.height}` : ''
-  const encodedURL = encodeURIComponent(url.replace(/'/g, '%27'))
-  const urlParam = `${encodedURL}`
-  const cacheURL = 'https://pocket-image-cache.com' //direct'
-  return `${cacheURL}/${resizeParam}/filters:no_upscale()/${urlParam}`
-}
-
```
Comment (anthony-liddle) [inline src/common/helpers.js:96]: We're using NONE of these! 🧹

File: src/common/interface.js [hunk 6]
```diff
@@ -141,15 +53,6 @@ export function inactiveIcon() {
...
 
-/* References
-–––––––––––––––––––––––––––––––––––––––––––––––––– */
-
-export function getExtensionInfo() {
-  return new Promise((resolve) => {
-    return chrome.management.getSelf(resolve)
-  })
-}
-
```
Comment (anthony-liddle) [inline src/common/interface.js:152]: All of this is garbage! 🧹

File: src/common/utilities.js [hunk 7]
```diff
@@ -88,7 +15,3 @@ export function getBool(value) {
     parseInt(value, 10) === 1
   )
 }
-
-export function mergeDedupe(arrayOfArrays) {
-  return [...new Set([].concat(...arrayOfArrays))]
-}
```
Comment (anthony-liddle) [inline src/common/utilities.js:94]: Trash functions 🧹

File: src/components/heading/heading.js [hunk 8]
```diff
@@ -9,20 +9,28 @@ const headingStyle = css`
...
 
   return (
     <header className={headingStyle}>
-      <PocketLogo />
-      <div className="saveBlock">{saveStatusCopy[saveStatus]}</div>
+      <div>
+        <PocketLogo />
+        <div className="saveBlock">{saveStatusCopy[saveStatus]}</div>
+      </div>
+      <button onClick={removeAction}>Remove</button>
```
Comment (anthony-liddle) [inline src/components/heading/heading.js:33]: Adding a temp button for Remove to be styled. It works though!

File: src/components/heading/heading.js [hunk 9]
```diff
@@ -9,20 +9,28 @@ const headingStyle = css`
...
-    remove: 'Removing...',
+    save_failed: 'Something went wrong!',
+    removing: 'Removing...',
     removed: 'Removed',
-    tags_saving: 'Saving tags...'
+    remove_failed: 'Something went wrong!',
+    tags_saving: 'Saving tags...',
+    tags_saved: 'Tags saved',
+    tags_failed: 'Something went wrong!',
+    tags_error: 'Tags limited to 25 characters'
```
Comment (anthony-liddle) [inline src/components/heading/heading.js:24]: We'll want to wrap all of these with the localize function

File: src/connectors/footer/footer.js [hunk 10]
```diff
@@ -2,7 +2,7 @@ import React from 'react'
 import { Footer } from 'components/footer/footer'
 
 export const FooterConnector = () => {
-  const settingsClick = () => console.log('settings clicked')
+  const settingsClick = () => console.log('settings clicked') //chrome.runtime.openOptionsPage()
```
Comment (anthony-liddle) [inline src/connectors/footer/footer.js:5]: putting this here for safe keeping until it's hooked up.

File: src/connectors/heading/heading.js [hunk 11]
```diff
@@ -35,6 +49,24 @@ export const HeadingConnector = () => {
...
+      case TAG_SYNC_FAILURE: {
+        return setSaveStatus('tags_failed')
+      }
+
+      case UPDATE_ITEM_PREVIEW: {
+        const { item } = payload
+        setItemId(item?.itemId)
+        return
+      }
+
```
Comment (anthony-liddle) [inline src/connectors/heading/heading.js:69]: We'll be adding one more case for when there's an error related to tagging.

File: src/manifest.json [hunk 12]
```diff
@@ -43,14 +43,6 @@
...
         "pages/logout.js"
       ]
-    },
-    {
-      "matches": [
-        "*://twitter.com/*"
-      ],
-      "js": [
-        "pages/twitter.js"
-      ]
```
Comment (anthony-liddle) [inline src/manifest.json:53]: 🎉

File: src/pages/background/userActions.js [hunk 13]
```diff
@@ -95,33 +90,16 @@ async function save({ linkUrl, pageUrl, title, tabId }) {
...
-//   chrome.tabs.sendMessage(tabId, { action: ARCHIVE_ITEM_REQUEST })
-
-//   const { response } = await archiveItem(item_id, actionInfo)
-//   const message = response
-//     ? { action: ARCHIVE_ITEM_SUCCESS, payload }
-//     : { action: ARCHIVE_ITEM_FAILURE, payload }
-
-//   chrome.tabs.sendMessage(tabId, message)
-// }
-
```
Comment (anthony-liddle) [inline src/pages/background/userActions.js:114]: Since archiving from the extension is going away...

File: src/pages/injector/app.js [hunk 14]
```diff
@@ -5,10 +5,32 @@ import { ItemPreviewConnector } from 'connectors/item-preview/item-preview'
...
+      case SAVE_TO_POCKET_REQUEST: {
+        return setIsOpen(true)
+      }
+
+      default: {
+        return
+      }
+    }
+  }
+
```
Comment (anthony-liddle) [inline src/pages/injector/app.js:33]: We needed to add this back for instances when a user clicks Save twice on the page

**Review by wtfluckey — APPROVED** [review]
Byeeeee ✌️

**Pull request merged**